### PR TITLE
Bug/187 track changes from embedded documents (NOT deeply nested)

### PR DESCRIPTION
**Pull request by Startouf**
I finished a set of changes for tracking embeds_one changes from the child

It only works on the first level of a nested relationship (won't work for deeply nested relationships, I have added a "pending" test regarding this 

I'm using the trick of "nested keys with dots" so that I can more easily make the difference between a nested document and just a field of type hash, so as to improve the tracking record

Fixes partially https://github.com/mongoid/mongoid-history/issues/187

### CODE REVIEW

**Comment by dblock** [comment]
For rubocop, `rubocop -a ; rubocop --auto-gen-config`. Will do a proper CR when 🍏, looks promising!

**Comment by Startouf** [comment]
@dblock build is green, waiting for your CR

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ### 0.8.2 (Next)
 
+* [#232](https://github.com/mongoid/mongoid-history/pull/232): Bug/187 track changes from embedded documents (not deeply nested) - [@Startouf](https://github.com/Startouf).
```
Comment (dblock) [inline CHANGELOG.md]: By convention we just do `[#232], [#187]: ...`, but I won't hold the CR for this.

**Comment by dblock** [comment]
I'm merging it. 👏

**Pull request merged**